RecoTracker/MkFit/plugins/MkFitProducer.cc

```diff
@@ -38,67 +49,84 @@ class MkFitProducer : public edm::global::EDProducer<edm::StreamCache<mkfit::MkB
 private:
   void produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const override;
 
-  edm::EDGetTokenT<MkFitInputWrapper> hitsSeedsToken_;
-  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
+  edm::EDGetTokenT<MkFitHitWrapper> hitToken_;
+  edm::EDGetTokenT<MkFitSeedWrapper> seedToken_;
+  edm::EDGetTokenT<edm::ContainerMask<edmNew::DetSetVector<SiPixelCluster> > > pixelMaskToken_;
+  edm::EDGetTokenT<edm::ContainerMask<edmNew::DetSetVector<SiStripCluster> > > stripMaskToken_;
+  edm::ESGetToken<MkFitGeometry, TrackerRecoGeometryRecord> mkFitGeomToken_;
   edm::EDPutTokenT<MkFitOutputWrapper> putToken_;
   std::function<double(mkfit::Event&, mkfit::MkBuilder&)> buildFunction_;
+  const float minGoodStripCharge_;
+  int iterationNumber_;  // TODO: temporary solution
+  bool seedCleaning_;
   bool backwardFitInCMSSW_;
+  bool removeDuplicates_;
   bool mkFitSilent_;
+  bool limitConcurrency_;
 };
 
 MkFitProducer::MkFitProducer(edm::ParameterSet const& iConfig)
-    : hitsSeedsToken_{consumes<MkFitInputWrapper>(iConfig.getParameter<edm::InputTag>("hitsSeeds"))},
-      geomToken_{esConsumes<TrackerGeometry, TrackerDigiGeometryRecord>()},
+    : hitToken_{consumes<MkFitHitWrapper>(iConfig.getParameter<edm::InputTag>("hits"))},
+      seedToken_{consumes<MkFitSeedWrapper>(iConfig.getParameter<edm::InputTag>("seeds"))},
+      mkFitGeomToken_{esConsumes<MkFitGeometry, TrackerRecoGeometryRecord>()},
       putToken_{produces<MkFitOutputWrapper>()},
+      minGoodStripCharge_{static_cast<float>(
+          iConfig.getParameter<edm::ParameterSet>("minGoodStripCharge").getParameter<double>("value"))},
+      iterationNumber_{iConfig.getParameter<int>("iterationNumber")},
+      seedCleaning_{iConfig.getParameter<bool>("seedCleaning")},
       backwardFitInCMSSW_{iConfig.getParameter<bool>("backwardFitInCMSSW")},
-      mkFitSilent_{iConfig.getUntrackedParameter<bool>("mkFitSilent")} {
+      removeDuplicates_{iConfig.getParameter<bool>("removeDuplicates")},
+      mkFitSilent_{iConfig.getUntrackedParameter<bool>("mkFitSilent")},
+      limitConcurrency_{iConfig.getUntrackedParameter<bool>("limitConcurrency")} {
+  const auto clustersToSkip = iConfig.getParameter<edm::InputTag>("clustersToSkip");
+  if (not clustersToSkip.label().empty()) {
+    pixelMaskToken_ = consumes(clustersToSkip);
+    stripMaskToken_ = consumes(clustersToSkip);
+  }
+
   const auto build = iConfig.getParameter<std::string>("buildingRoutine");
-  bool isFV = false;
   if (build == "bestHit") {
-    buildFunction_ = mkfit::runBuildingTestPlexBestHit;
+    //buildFunction_ = mkfit::runBuildingTestPlexBestHit;
+    throw cms::Exception("Configuration") << "bestHit is temporarily disabled";
   } else if (build == "standard") {
-    buildFunction_ = mkfit::runBuildingTestPlexStandard;
+    //buildFunction_ = mkfit::runBuildingTestPlexStandard;
+    throw cms::Exception("Configuration") << "standard is temporarily disabled";
```

Comment:
Should we keep the placeholders to (eventually) use `bestHit` and `standard` builders in addition to the `cloneEngine`?

Reply:
Thanks, I'll leave the exceptions there then (until `mkfit::run_OneIteration` provides some way for using different builder)